protocols/delegate-manager/contracts/DelegateManager.sol

```diff
@@ -0,0 +1,121 @@
+/*
+  Copyright 2019 Swap Holdings Ltd.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity 0.5.12;
+pragma experimental ABIEncoderV2;
+
+import "@airswap/indexer/contracts/interfaces/IIndexer.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
+import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
+import "@airswap/delegate/contracts/interfaces/IDelegate.sol";
+import "@airswap/structs/contracts/Structs.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+contract DelegateManager is Ownable {
+
+    IDelegateFactory public factory;
+    IDelegate public delegate;
+
+    /**
+      * @notice Constructor to create the Delegate Manager
+      * @dev factory is unable to be changed after the DelegateManager is constructed
+      * @param _factory address of the factory contract that will create the delegates
+      * @param _tradeWallet the wallet that the delegate will be trading on behalf of
+      */
+    constructor(IDelegateFactory _factory, address _tradeWallet) public {
+        factory = _factory;
+        delegate = IDelegate(factory.createDelegate(msg.sender, _tradeWallet));
```

Comment:
![Screen Shot 2019-10-22 at 10 16 52 AM](https://user-images.githubusercontent.com/3590816/67295439-1b8d1700-f4b5-11e9-9e34-90065dc81a4d.png)

Reply:
That would require the Delegate knowing about the indexer though which I'm less keen on. This could almost be combines with the other comment on the PR that alters the factory to deploy the manager too. You could have the following:
`Factory.deployDelegateAndManager`, which performs:
- Deploy manager with msg.sender as the owner
- Deploy delegate with `manager` as the owner

Then in one step all the permissions are setup.

Reply:
> That would require the Delegate knowing about the indexer
Only in context to `setRuleAndIntent` and `unsetRuleAndIntent`. Removing the manager puts the delegate in control of managing itself.

Reply:
> If we do this it means the DelegateFactory would do more than create delegates.
Yes. But it can just be a slightly more generic factory - for creating Delegates and Managers for Delegates. If we are assuming that people who get Delegates might want a Manager for it too, it makes complete sense to have a function where that all gets set up for them - instead of them having to deploy them separately and then change permissions in them to let them communicate. One function could do all of that.